examples/seismic/viscoelastic/wavesolver.py

```diff
@@ -0,0 +1,133 @@
+            The computed vertical particle velocity.
+        txx : TimeFunction, optional
+            The computed horizontal stress.
+        tzz : TimeFunction, optional
+            The computed vertical stress.
+        txz : TimeFunction, optional
+            The computed diagonal stresss.
+        rxx: TimeFunction, optional
+            The computed horizontal memory variable.
+        rzz: TimeFunction, optional
+            The computed vertical memory variable.
+        rxz: TimeFunction, optional
+            The computed diagonal memory variable.
+        vp : Function, optional
+            The time-constant P-wave velocity (km/s).
+        qp : Function, optional
+            The P-wave quality factor (dimensionless).
+        vs : Function, optional
+            The time-constant S-wave velocity (km/s).
+        qs : Function, optional
+            The S-wave quality factor (dimensionless)
+        rho : Function, optional
+            The time-constant density (rho=1 for water).
+        save : int or Buffer, optional
+            Option to store the entire (unrolled) wavefield.
+
+        Returns
+        -------
+        Rec1 (txx), Rec2 (tzz), particle velocities vx and vz, stress txx,
+        tzz and txz, memory variables rxx, rzz, rxz and performance summary.
+        """
+        # Source term is read-only, so re-use the default
+        src = src or self.geometry.src
+        # Create a new receiver object to store the result
+        rec1 = rec1 or Receiver(name='rec1', grid=self.model.grid,
+                                time_range=self.geometry.time_axis,
+                                coordinates=self.geometry.rec_positions)
+        rec2 = rec2 or Receiver(name='rec2', grid=self.model.grid,
+                                time_range=self.geometry.time_axis,
+                                coordinates=self.geometry.rec_positions)
+
+        # Create all the fields vx, vz, tau_xx, tau_zz, tau_xz, r_xx, r_zz, r_xz
+        save_t = src.nt if save else None
+        vx, vy, vz = vector_function('v', self.model, save_t, self.space_order)
+        txx, tyy, tzz, txy, txz, tyz = tensor_function('t', self.model, save_t,
+                                                       self.space_order)
+        rxx, ryy, rzz, rxy, rxz, ryz = tensor_function('r', self.model, save_t,
+                                                       self.space_order)
+        kwargs['vx'] = vx
+        kwargs['vz'] = vz
+        kwargs['txx'] = txx
+        kwargs['tzz'] = tzz
+        kwargs['txz'] = txz
+        kwargs['rxx'] = rxx
+        kwargs['rzz'] = rzz
+        kwargs['rxz'] = rxz
+        if self.model.grid.dim == 3:
+            kwargs['vy'] = vy
+            kwargs['tyy'] = tyy
+            kwargs['txy'] = txy
+            kwargs['tyz'] = tyz
+            kwargs['ryy'] = ryy
+            kwargs['rxy'] = rxy
+            kwargs['ryz'] = ryz
+        # Pick m from model unless explicitly provided
```

Comment:
vp* insterad of m?

Reply:
and not just `vp`. I think it's copy-paste leftover

Reply:
Yeah, was a copy and paste relic here and in elastic. Changed to 'physical parameters'.